src/loaders/lottie/tvgLottieProperty.h

```diff
@@ -516,35 +685,45 @@ struct LottiePathSet : LottieProperty
         auto s = frame->value.pts;
         auto e = (frame + 1)->value.pts;
 
-        if (roundness > ROUNDNESS_EPSILON) {
-            auto interpPts = (Point*)malloc(frame->value.ptsCnt * sizeof(Point));
-            auto p = interpPts;
-            for (auto i = 0; i < frame->value.ptsCnt; ++i, ++s, ++e, ++p) {
-                *p = lerp(*s, *e, t);
-                if (transform) *p *= *transform;
-            }
-            _modifier(interpPts, frame->value.ptsCnt, frame->value.cmds, frame->value.cmdsCnt, cmds, pts, nullptr, roundness);
-            free(interpPts);
-            return true;
-        } else {
+        if (roundness <= ROUNDNESS_EPSILON && tvg::zero(offset)) {
             for (auto i = 0; i < frame->value.ptsCnt; ++i, ++s, ++e) {
                 auto pt = lerp(*s, *e, t);
                 if (transform) pt *= *transform;
                 pts.push(pt);
             }
             _copy(&frame->value, cmds);
+            return true;
+        }
+
+        auto interpPts = (Point*)malloc(frame->value.ptsCnt * sizeof(Point));
+        auto p = interpPts;
+        for (auto i = 0; i < frame->value.ptsCnt; ++i, ++s, ++e, ++p) {
+            *p = lerp(*s, *e, t);
+            if (transform) *p *= *transform;
+        }
+
+        if (roundness > ROUNDNESS_EPSILON) {
+            if (!tvg::zero(offset)) {
+                Array<PathCommand> cmds1;
+                Array<Point> pts1;
+                _modifier(interpPts, frame->value.ptsCnt, frame->value.cmds, frame->value.cmdsCnt, cmds1, pts1, nullptr, roundness);
+                _modifier(pts1.data, pts1.count, cmds1.data, cmds1.count, cmds, pts, offset, join, miter);
+            } else _modifier(interpPts, frame->value.ptsCnt, frame->value.cmds, frame->value.cmdsCnt, cmds, pts, nullptr, roundness);
+        } else if (!tvg::zero(offset)) {
+            _modifier(interpPts, frame->value.ptsCnt, frame->value.cmds, frame->value.cmdsCnt, cmds, pts, offset, join, miter);
         }
+        free(interpPts);
         return true;
     }
 
 
-    bool operator()(float frameNo, Array<PathCommand>& cmds, Array<Point>& pts, Matrix* transform, float roundness, LottieExpressions* exps)
+    bool operator()(float frameNo, Array<PathCommand>& cmds, Array<Point>& pts, Matrix* transform, float roundness, float offset, StrokeJoin join, float miter, LottieExpressions* exps)
```

Comment:
There are too many parameters for an unusual feature. These may need to be unified into a single parameter as a modifier data something like a `PathModifier`

```
 float roundness, float offset, StrokeJoin join, float miter,
```